core/src/scene/sceneLoader.cpp

```diff
@@ -954,29 +975,44 @@ void SceneLoader::loadSource(const std::shared_ptr<Platform>& platform, const st
     }
 
     // distinguish tiled and non-tiled sources by url
-    bool tiled = url.find("{x}") != std::string::npos &&
+    bool tiled = url.size() > 0 &&
+        url.find("{x}") != std::string::npos &&
         url.find("{y}") != std::string::npos &&
         url.find("{z}") != std::string::npos;
 
+    bool isMBTilesFile = false;
+    {
+        const char* extStr = ".mbtiles";
+        const size_t extLength = strlen(extStr);
+        const size_t urlLength = url.length();
+        isMBTilesFile = urlLength > extLength && (url.compare(urlLength - extLength, urlLength, extStr) == 0);
```

Comment:
Hmm I think this should be:
`url.compare(urlLength - extLength, extLength, extStr) == 0`

To check if substring [arg1, arg1+arg2) in `url` is same as arg3!